lib/login_page.dart

```diff
@@ -0,0 +1,416 @@
+  final _formKey = GlobalKey<FormState>();
+  TextEditingController usernameController = TextEditingController();
+  TextEditingController passwordController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    mp['eng'] = "English";
+    mp['ara'] = "العربية";
+    mp['fre'] = "Français";
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    isMobile = MediaQuery.of(context).orientation == Orientation.portrait;
+    double h = ScreenUtil().screenHeight;
+    double w = ScreenUtil().screenWidth;
+    return isLoggedIn
+        ? RegistrationClient(
+            onLogout: () {
+              setState(() {
+                username = '';
+                isUserValidated = false;
+                isLoggedIn = false;
+              });
+            },
+          )
+        : SafeArea(
+            child: Scaffold(
+              backgroundColor: Utils.appSolidPrimary,
+              body: Stack(
+                children: [
+                  Positioned(
+                    bottom: 0,
+                    left: 16.w,
+                    child: _getBuildingsImage(),
+                  ),
+                  Container(
+                    height: h,
+                    width: w,
+                    child: SingleChildScrollView(
+                      child: Column(
+                        children: [
+                          _appBarComponent(),
+                          SizedBox(
+                            height: isMobile ? 50.h : 132.h,
+                          ),
+                          Container(
+                            padding: EdgeInsets.symmetric(
+                              horizontal: isMobile ? 16.w : 80.w,
+                            ),
+                            child: isMobile ? _mobileView() : _tabletView(),
+                          ),
+                        ],
+                      ),
+                    ),
+                  ),
+                ],
+              ),
+            ),
+          );
+  }
+
+  Future<void> _login(String username, String password) async {
+    String response;
+    Map<String, dynamic> mp;
+    try {
+      response = await platform
+          .invokeMethod("login", {'username': username, 'password': password});
+      mp = jsonDecode(response);
+      debugPrint('Login: $mp');
+    } on PlatformException catch (e) {
+      mp = {
+        'login_response': 'Login Failed..Try Again!',
+        'isLoggedIn': 'false',
+      };
+    }
+    setState(() {
+      loginResponse = mp['login_response'];
+      isLoggedIn = mp['isLoggedIn'] == 'true' ? true : false;
+    });
+  }
+
+  Future<void> _validateUsername() async {
+    String response;
+    Map<String, dynamic> mp;
+
+    try {
+      response = await platform
+          .invokeMethod("validateUsername", {'username': username});
+      mp = jsonDecode(response);
+    } on PlatformException catch (e) {
+      mp = {
+        'user_response': 'Failed..Try Again!',
```

Comment:
can we create a standard error response to the UI, and also we should return error code. and the message should be taken from i18n bundle

Reply:
Created a login response model. Where error code is hardcoded for now and added messages in the l10n files.